// @gradientai/nodejs-sdk/dist/esm/model/modelAdapter.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.ModelAdapter = void 0;
const abstractModel_1 = require("./abstractModel");
class ModelAdapter extends abstractModel_1.Model {
    constructor({ apiInstance, id, baseModelId, name, workspaceId, }) {
        super({ apiInstance, id, workspaceId });
        this.delete = () => __awaiter(this, void 0, void 0, function* () {
            yield this.apiInstance.deleteModel({
                id: this.id,
                xGradientWorkspaceId: this.workspaceId,
            });
        });
        this.fineTune = ({ samples, }) => __awaiter(this, void 0, void 0, function* () {
            const { data: { numberOfTrainableTokens, sumLoss }, } = yield this.apiInstance.fineTuneModel({
                id: this.id,
                xGradientWorkspaceId: this.workspaceId,
                fineTuneModelBodyParams: { samples },
            });
            return { numberOfTrainableTokens, sumLoss };
        });
        this.baseModelId = baseModelId;
        this.name = name;
    }
}
exports.ModelAdapter = ModelAdapter;
