// @gradientai/nodejs-sdk/dist/cjs/model/abstractModel.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Model = void 0;
class Model {
    constructor({ apiInstance, id, workspaceId, }) {
        this.complete = ({ autoTemplate, guidance, maxGeneratedTokenCount, query, rag, temperature, topK, topP, }) => __awaiter(this, void 0, void 0, function* () {
            const { data: { finishReason, generatedOutput }, } = yield this.apiInstance.completeModel({
                id: this.id,
                xGradientWorkspaceId: this.workspaceId,
                completeModelBodyParams: {
                    autoTemplate,
                    guidance,
                    query,
                    rag,
                    maxGeneratedTokenCount,
                    temperature,
                    topK,
                    topP,
                },
            });
            return { finishReason, generatedOutput };
        });
        this.apiInstance = apiInstance;
        this.id = id;
        this.workspaceId = workspaceId;
    }
}
exports.Model = Model;
